### ISSUE TITLE
Mailer docker image is failing when deploying

### ISSUE THREAD

**Issue by leoramirez-mulesoft** [cloud-custodian/cloud-custodian#5555]
We are using `cloudcustodian/mailer:latest`, but getting errors below:

```
15:31:39 + c7n-mailer --config mailer.yaml --update-lambda -t templates
15:31:41 Traceback (most recent call last):
15:31:41   File "/usr/local/bin/c7n-mailer", line 11, in <module>
15:31:41     load_entry_point('c7n-mailer', 'console_scripts', 'c7n-mailer')()
15:31:41   File "/src/tools/c7n_mailer/c7n_mailer/cli.py", line 237, in main
15:31:41     deploy.provision(mailer_config, functools.partial(session_factory, mailer_config))
15:31:41   File "/src/tools/c7n_mailer/c7n_mailer/deploy.py", line 99, in provision
15:31:41     archive = get_archive(config)
15:31:41   File "/src/tools/c7n_mailer/c7n_mailer/deploy.py", line 61, in get_archive
15:31:41     archive = PythonPackageArchive(modules=deps)
15:31:41   File "/src/c7n/mu.py", line 85, in __init__
15:31:41     self.add_modules(None, modules)
15:31:41   File "/src/c7n/mu.py", line 129, in add_modules
15:31:41     module = importlib.import_module(module_name)
15:31:41   File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
15:31:41     return _bootstrap._gcd_import(name[level:], package, level)
15:31:41   File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
15:31:41   File "<frozen importlib._bootstrap>", line 983, in _find_and_load
15:31:41   File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
15:31:41 ModuleNotFoundError: No module named 'simplejson'
```

Seems that `latest` image is missing some modules.

**Comment by JohnHillegass** [comment]
Hey @leoramirez-mulesoft, we upgraded the python version in that package to v3.x a few days ago and simplejson is no longer needed as a dependency for datadog. It was an oversight in the testing for it. Thanks for reporting. I'll submit a PR to address it shortly.

**Comment by JohnHillegass** [comment]
Should be resolved when #5521 is merged.

**Comment by kapilt** [comment]
@JohnHillegass  fwiw, this is unrelated to the update 3.7 lambda runtime.

afaics this issue dates back to switching over to poetry and redoing the docker images, as we don't have simplejson as part of the transitive dependency graph anymore. simply removing it from mu.py suffices.
